Docker-compose/docker-compose.yml

```diff
@@ -1,55 +1,57 @@
-version: '3'
 services:
   satosa-mongo:
     image: mongo
     container_name: satosa-mongo
-    restart: always
     environment:
       MONGO_INITDB_DATABASE: oidcop
       MONGO_INITDB_ROOT_USERNAME: "${MONGO_DBUSER:-satosa}"
       MONGO_INITDB_ROOT_PASSWORD: "${MONGO_DBPASSWORD:-thatpassword}"
+      TZ: "${TZ:-Europe/Rome}"
     volumes:
-      - mongodata:/data/db
-      - /usr/share/zoneinfo/Europe/Rome:/etc/localtime:ro
+      - ./mongo/db:/data/db
       - ./mongo/init-mongo.sh:/docker-entrypoint-initdb.d/init-mongo.sh
+      # - mongodata:/data/db # to be used for external volumes
     ports:
       - '27017-27019:27017-27019'
     networks:
       - satosa-saml2spid
 
   satosa-mongo-express:
+    profiles:
+      - demo
+      - dataentry
     image: mongo-express
     container_name: satosa-mongo-express
-    restart: always
     ports:
       - 8082:8081
+    depends_on:
+      - satosa-mongo
     environment:
       ME_CONFIG_BASICAUTH_USERNAME: satosauser
```

Comment:
Wouldnt be mettere to have them in ENV?

Reply:
For instances with only spid-saml backend? I can try :+1:

Reply:
I have tried and I can't use an Environment to enable mongo dependence on compose because `depends_on` don't accept "null" as value:
`service "satosa-saml2spid" depends on undefined service "": invalid` 

I can propose a new docker compose profile for MongoDB? The profiles can be:

**default (without profile):**
* satosa-sa,l2spid (without mongo dependancy)
* satosa-nginx

**db:**
* *default
*  satosa-mongo

**dataentry:**
* *default
* *db
* satosa-mongoexpress

**dev:**
* *default
* django-sp
* spid-samlcheck

**demo:**
* *default**
* *db
* *dataentry
*  *dev

If I use an external MongoDB, is sufficient don't start the `db` profile (only default) or use `dev` profile for a full system without mongo.
What do you think?